### DISCUSSION TITLE
Systematically extracting data: Writing raw ES queries?

### DISCUSSION THREAD

**Discussion by Joost385** [komoot/photon#575] [Q&A] [answered]
First of all thank you for this awesome project :)

I would like to systematically extract data, like cities, towns, villages, post codes ...
The provided search api does not seem to be made for that purpose.
Is there any way to write raw ES queries? I know that the api is exposed on port 2322 is there a port for ES as well?

**Comment by lonvia** [comment, marked as answer]
Photon runs ElasticSearch in embedded mode using the old-style API. there is no HTTP REST API endpoint. However, you can simply take the photon data and set it as data directory on a standard ElasticSerach 5.5 installation. This then gives you the standard ES API.

That said, I strongly recommend that you use Nominatim for systematic data extraction instead. The data there is much more complete (e.g. full geometries instead of centroids), it already comes with an 'export' command for the simple cases of CSV export and if that isn't enough, you have full access to the Postgresql data table to combine whatever you need.

**Reply by Joost385** [reply]
Thank you for pointing this out, all the different projects around osm can be confusing :)
I just found this wonderful repo (mediagis/nominatim-docker) which I will use to set everything up.

Have a great day!
